File: Week_2/485_MaxConsecutiveOnes.cpp
#include<iostream>
#include<stdio.h>
#include<unordered_map>
#include<string>
#include<vector>
using namespace std;

int findMaxConsecutiveOnes(vector<int>& nums) {
    int curr = 0;
    int max_cnt = 0;
    for (int r = 0; r < nums.size(); r++) {
        if (nums[r] == 1) {
            curr++;
        } else {
            curr = 0;
        }
        max_cnt = max(max_cnt, curr);
    }
    return max_cnt;
}

File: Week_2/485_MaxConsecutiveOnes.py
from typing import List
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        ctr = []
        ctr_i = 0
        for i in nums:
            if i == 1:
                ctr_i += 1
            else:
                ctr.append(ctr_i)
                ctr_i = 0
        ctr.append(ctr_i)
        return max(ctr)

## counts 1s, add 1 to a counter list when 0 is encountered, find max in the counter list

import re
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        strin = ''.join([str(i) for i in nums])
        rege = r"1{0,}"
        matches = re.findall(rege, strin)
        matches = [len(i) for i in matches]
        return max(matches)
# using regex

